test/corecel/random/RanluxppRngEngine.test.cc

```diff
@@ -0,0 +1,236 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/random/Ranluxpp.test.cc
+//---------------------------------------------------------------------------//
+#include "corecel/random/engine/RanluxppRngEngine.hh"
+
+#include <memory>
+
+#include "corecel/data/CollectionStateStore.hh"
+#include "corecel/random/data/RanluxppRngData.hh"
+#include "corecel/random/data/RanluxppTypes.hh"
+#include "corecel/random/distribution/GenerateCanonical.hh"
+#include "corecel/random/params/RanluxppRngParams.hh"
+
+#include "RngTally.hh"
+#include "celeritas_test.hh"
+
+namespace celeritas
+{
+namespace test
+{
+//---------------------------------------------------------------------------//
+class RanluxppRngEngineTest : public Test
+{
+  protected:
+    using HostStore
+        = CollectionStateStore<celeritas::RanluxppRngStateData, MemSpace::host>;
+    using DeviceStore
+        = CollectionStateStore<celeritas::RanluxppRngStateData, MemSpace::device>;
+    using uint_t = RanluxppUInt;
+
+    void SetUp() override
+    {
+        params_ = std::make_shared<RanluxppRngParams>(12345);
+    }
+
+    std::shared_ptr<RanluxppRngParams> params_;
+};
+
+TEST_F(RanluxppRngEngineTest, host)
+{
+    // Construct and initialize
+    HostStore states(params_->host_ref(), StreamId{0}, 8);
+
+    Span<RanluxppRngState> state_ref
+        = states.ref().state[AllItems<RanluxppRngState>{}];
+
+    // Check that initial states are reproducibly random by reading the data as
+    // a raw array of uints
+    std::vector<uint_t> flattened(10 * 8);
+    //(9 * sizeof(RanluxppUInt) + sizeof(unsigned int) + sizeof(int)) * 8);
+    ASSERT_EQ(flattened.size() * sizeof(uint_t),
+              state_ref.size() * sizeof(RanluxppRngState));
+    ASSERT_TRUE(std::is_standard_layout_v<RanluxppRngState>);
+    ASSERT_EQ(0, offsetof(RanluxppRngState, state));
+    ASSERT_EQ(9 * sizeof(RanluxppUInt), offsetof(RanluxppRngState, carry));
+    ASSERT_EQ(9 * sizeof(RanluxppUInt) + sizeof(unsigned int),
+              offsetof(RanluxppRngState, position));
+    std::copy_n(
+        &state_ref.begin()->state[0], flattened.size(), flattened.begin());
```

Comment:
Unfortunately this isn't legal for the Ranlux state; it's a shortcut I took for the Xorwow since it's several identically-sized integers. Ranlux has 64-bit and 32-bit mixed together. Try:

```suggestion
    // Check that initial states are reproducibly random
    std::vector<RanluxppUInt> flattened;
    for (RanluxppRngState const& s : state_ref)
    {
        flattened.insert(flattened.end(), s.state.begin(), s.state.end());
        flattened.push_back(s.carry);
        flattened.push_back(s.position);
    }
```

Reply:
Updated test.